src/libcore/fmt/num.rs

```diff
@@ -83,6 +81,8 @@ trait GenericRadix {
             }
         }
         let buf = &buf[curr..];
+        // SAFETY: The only chars in `buf` are created by `Self::digit` which are assumed to be
+        // valid UTF-8
         let buf = unsafe {
             str::from_utf8_unchecked(slice::from_raw_parts(MaybeUninit::first_ptr(buf), buf.len()))
```

Comment:
This is not safe, I believe, as the first_ptr function's pointer is only valid for the first element of the array (i.e., `&s[0]` cannot then be used to get to `[2]` or whatever). cc @RalfJung, perhaps @Centril -- am I right about this rule?

We should either update the code to use `MaybeUninit::slice_get_ref` or add a FIXME to the safety comment.

Generally speaking, it's a bug if the SAFETY comment doesn't mention *something* about the unsafe calls/expressions within the block; here it does not document why the buf is valid for it's entire length, for example.

Reply:
I think this is right, but I'll leave it to @RalfJung to confirm. At any rate, I think enhancing the docs on `first_ptr` would be a good idea.

Reply:
`first_ptr_mut` does the right thing internally to create a pointer that may be sued for the entire slice.

The methods correspond exactly to `slice::as_ptr`... probably they could be named better to reflect that better.

Reply:
We should at least document that then on the methods, thanks for the clarifications!

Resolving.